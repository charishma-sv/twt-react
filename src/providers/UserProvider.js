import React, { Component, createContext } from 'react';
import { auth, getUserDocument } from '../firebase';
export const UserContext = createContext({ user: null });
class UserProvider extends Component {
  state = {
    user: null,
  };

  componentDidMount = async () => {
    auth.onAuthStateChanged(async (userAuth) => {
      const user = await getUserDocument(userAuth);
      console.log('user in usr provider', user);
      this.setState({ user });
    });
  };
  updateUser = (user) => {
    this.setState({ user });
  };
  render() {
    return (
      <UserContext.Provider
        value={{ user: this.state.user, updateUser: this.updateUser }}
      >
        {this.props.children}
      </UserContext.Provider>
    );
  }
}
export default UserProvider;
