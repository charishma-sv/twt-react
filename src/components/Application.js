import React, { useContext } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import HomePage from './HomePage';
import Login from './Login';
import Feed from './Feed';
import { UserContext } from '../providers/UserProvider';
import ProtectedRoute from './Routes/ProtectedRoute';
import PublicRoute from './Routes/PublicRoute';
import Signup from './Signup';
const Application = () => {
  const { user, updateUser } = useContext(UserContext);
  return (
    <Router>
      <div>
        <PublicRoute component={Login} user={user} path="/login" />
        <PublicRoute
          component={Signup}
          user={user}
          updateUser={updateUser}
          path="/signup"
        />
        <ProtectedRoute component={Feed} user={user} path="/feed" />
        <PublicRoute component={HomePage} user={user} path="/" exact />
      </div>
    </Router>
  );
};
export default Application;
