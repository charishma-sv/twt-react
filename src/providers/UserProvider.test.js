import React from 'react';
import { act, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Application from '../components/Application';
import UserProvider from './UserProvider';

describe('User provider snapshot when user is null', () => {
  it('should render LoadingComponent', () => {
    const root = document.createElement('div');
    document.body.appendChild(root);
    let user = null;
    let tree = renderer.create(
      <UserProvider value={user}>
        <Application />
      </UserProvider>,
      root
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
describe('User provider snapshot when user is undefined', () => {
  it('should render Homepage component', () => {
    let user = jest.fn(() => Promise.resolve(undefined));
    const root = document.createElement('div');
    document.body.appendChild(root);
    const tree = renderer.create(
      <UserProvider value={user}>
        <Application />
      </UserProvider>,
      root
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
