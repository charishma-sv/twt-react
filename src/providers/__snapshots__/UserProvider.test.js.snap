// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User provider snapshot when user is null should render LoadingComponent 1`] = `
<div>
  <div
    className=" vh-100 d-flex justify-content-center align-items-center container"
  >
    <div
      className="spinner-grow"
    />
  </div>
</div>
`;

exports[`User provider snapshot when user is undefined should render Homepage component 1`] = `
<div>
  <div
    className=" vh-100 d-flex justify-content-center align-items-center container"
  >
    <div
      className="spinner-grow"
    />
  </div>
</div>
`;
